src/commands/bundle_for_gem.yml

```diff
@@ -8,10 +8,17 @@ parameters:
   project:
     type: string
 steps:
+  # md5sum's Gemfile and << parameters.project >>.gemspec (optionally, if it exists) to generate a unique
+  # cache key representing the contents of both files.
+  - run:
+      name: Generate a cache key that optionally includes a .gemspec file
+      command: |
+        echo $(find . -d 1 -type f \( -name Gemfile -o -name << parameters.project >>.gemspec \) -exec md5sum {} \; | md5sum | cut -d' ' -f1) >> BUNDLE_CACHE_KEY
+
   - restore_cache:
       name: Restore bundle from cache
       keys:
-        - v1-bundle-{{ checksum "Gemfile" }}--{{ checksum "<< parameters.project >>.gemspec" }}-<< parameters.ruby_version >>
+        - v1-bundle-{{ checksum "BUNDLE_CACHE_KEY" }}-<< parameters.ruby_version >>
```

Comment:
Is checksum here unnecessary since md5sum is being run above when creating BUNDLE_CACHE_KEY?